### DISCUSSION TITLE
Handle imputation to increase variant overlap

### DISCUSSION THREAD

**Discussion by Fiwx** [PGScatalog/pgsc_calc#146] [Ideas]
### Description of feature

It would be great if it could automatically handle imputation of the input files through Minimac4 or BEAGLE or similar.

**Comment by smlmbrt** [comment]
Hi @Fiwx, we agree but we think keeping the pgsc_calc separate from imputation might be better for a couple of reasons:

- Oftentimes the imputation reference panels can't be shared which limits the usability of implementing imputation. 
- We like that the pgsc_calc pipeline is more portable (you bring the pipeline to your data). Imputation pipelines tend to be more central (you upload your data to established imputation servers [e.g. Michigan, TopMed])
- Other imputation pipelines have been developed: https://github.com/genepi/imputationserver https://github.com/h3abionet/chipimputation

But you are correct that we could think more about the the recommended upstream steps for the pipeline - and happy to take suggestions on how to do that

**Reply by Fiwx** [reply]
Hi, that makes a lot of sense. Are there any considerations/differences of running imputed data through the pipeline? Everything I can think of is solved already: 
- Handling of imputation probability for the genotype field (use DS in samplesheet instead of GT).
- The fact that the reference panel may not be imputed but the input file(s) are I think this is solved ~fully by "identifying variants that are present in both the target genotypes and the reference panel."

I don't think the two considerations above are issues because the pipeline is very robust, but if you know of any best practices to follow when using imputed data, I would be interested to hear them.

Also, I have been assuming it is ideal (for most scores) to use imputed data to increase variant overlap. Do you agree with this notion?

**Reply by smlmbrt** [reply]
The pipeline actually sort of expects imputed genotypes as input as these are what will yield the highest number of matching positions between the PGS scoring file and the target genotypes. This is because the majority of genotyping data in cohort studies has been generated on chips and then imputed, leading to PGS developers mostly using these data and imputed variants for score development. The reference panel (1000 Genomes) isn't imputed that is correct, but since all the variants are genotypes it's used to be the most commonly used panel of genotypes itself.
